app/src/main/java/fr/neamar/kiss/result/TagDummyResult.java

```diff
@@ -0,0 +1,191 @@
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import fr.neamar.kiss.MainActivity;
+import fr.neamar.kiss.R;
+import fr.neamar.kiss.pojo.TagDummyPojo;
+import fr.neamar.kiss.utils.FuzzyScore;
+
+public class TagDummyResult extends Result {
+    private static final String TAG = TagDummyResult.class.getSimpleName();
+    private static Drawable gBackground = null;
+
+    TagDummyResult(@NonNull TagDummyPojo pojo) {
+        super(pojo);
+    }
+
+    private Drawable getShape(Context context, SharedPreferences sharedPreferences) {
+        boolean largeSearchBar = sharedPreferences.getBoolean("large-search-bar", false);
+        int barSize = context.getResources().getDimensionPixelSize(largeSearchBar ? R.dimen.large_bar_height : R.dimen.bar_height);
+
+        if ( gBackground == null || gBackground.getIntrinsicWidth() != barSize || gBackground.getIntrinsicHeight() != barSize ) {
+            int inset = (int)(3.f * context.getResources().getDisplayMetrics().density);   // 3dp to px
+            barSize -= 2 * inset;   // shrink size with the inset to keep the overall size
+
+            GradientDrawable shape = new GradientDrawable();
+            shape.setShape(GradientDrawable.RECTANGLE);
+            shape.setSize(barSize, barSize);
+            float rad = barSize / 2.3f;
+            shape.setCornerRadii(new float[]{rad, rad, rad, rad, rad, rad, rad, rad});
+            shape.setColor(0xFFffffff);
+
+            gBackground = new InsetDrawable(shape, inset);
+        }
+
+        return gBackground;
+    }
+
+    private Bitmap generateBitmap(Context context, SharedPreferences sharedPreferences) {
+        boolean largeSearchBar = sharedPreferences.getBoolean("large-search-bar", false);
+        int barSize = context.getResources().getDimensionPixelSize(largeSearchBar ? R.dimen.large_bar_height : R.dimen.bar_height);
+
+        int width, height = width = barSize;
+
+        // create a canvas from a bitmap
+        Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(bitmap);
+
+        // use StaticLayout to draw the text centered
+        TextPaint paint = new TextPaint();
+        paint.setFlags(Paint.ANTI_ALIAS_FLAG);
+        paint.setTextSize(.6f * height);
+
+        RectF rectF = new RectF(0, 0, width, height);
+        rectF.inset(1.f, 1.f);
+
+        // draw a white rounded background
+        paint.setColor(0xFFffffff);
+        canvas.drawRoundRect(rectF, width / 2.4f, height / 2.4f, paint);
+
+        int codepoint = pojo.getName().codePointAt(0);
+        String glyph = new String(Character.toChars(codepoint));
+        // If the codepoint glyph is an image we can't use SRC_IN to draw it.
+        boolean drawAsHole = true;
+        Character.UnicodeBlock block = null;
+        try {
+            block = Character.UnicodeBlock.of(codepoint);
+        } catch (IllegalArgumentException ignored) {
+        }
+        if (block == null)
```

Comment:
IMO this shouldn't be a gigantic if else, it should set the variable directly.

Reply:
directly to what? How can I avoid the `if else` tree?

Reply:
Maybe something like:

```
drawAsHole = !(block == null || "DINGBATS".equals(block.toString()) || "EMOTICONS".equals(block.toString()) || "MISCELLANEOUS_SYMBOLS".equals(block.toString()) || "MISCELLANEOUS_SYMBOLS_AND_PICTOGRAPHS".equals(block.toString()) || "SUPPLEMENTAL_SYMBOLS_AND_PICTOGRAPHS".equals(block.toString()) || "TRANSPORT_AND_MAP_SYMBOLS".equals(block.toString()));
```